src/regex.c

```diff
@@ -286,19 +287,22 @@ int do_regex_hybrid_crack(struct db_main *db, const char *regex,
 	}
 
 	while (c_iterator_next(iter)) {
-		c_iterator_value(iter, buffer);
-		c_simplestring_to_utf8_string(buffer, &word[0], sizeof(word));
 		c_simplestring_clear(buffer);
-		if (options.internal_cp != UTF_8)
-			utf8_to_cp_r(word, word, sizeof(word));
+		c_iterator_value(iter, buffer);
+		word = c_simplestring_to_string(buffer);
+    /**
+      * rexgen already creates the correct encoding
+      */
+		//if (options.internal_cp != UTF_8)
+		//	utf8_to_cp_r(word, word, sizeof(word));
 		if (options.mask) {
```

Comment:
I think the conversions you commented out worked as intended. That is, we do one of these things:
* Set UTF-8 locale and feed UTF-8 to librexgen, obviously expecting UTF-8 back.
* Set C locale and convert to some internal (8-bit but not multibyte) encoding before calling librexgen. In this case we expect the same encoding back (just transparent handling of the 8-bit chars).